### ISSUE TITLE
Allow to re-simulate a failed simulation

### ISSUE THREAD

**Issue by OpenModelica-TracImporter** [OpenModelica/OpenModelica#5472]

**Comment by casella** [comment]
@adeas31 this would be really convenient for debugging. I often deal with big, complex models. If the simulation fails because of bad solver settings, you need to go through the whole compilation process again and again, which is frustrating.

Can you deal with that in 1.26.0?

**Comment by casella** [comment]
@adeas31 last call, can we consider this for 1.26.0?

**Comment by casella** [comment]
@adeas31 we are debugging Buildings models this morning that fail at runtime. It's really, really bad and completely unnecessary that you have to re-compile the model every time, which maybe takes 5 minutes.

Can we do something about this?

Thanks!

**Comment by adeas31** [comment]
If the simulation has failed then re-simulating the same model will fail again. So the question is, do you just want to change the simulation settings or do you want to change model parameters or both before doing a re-simulate?

**Comment by casella** [comment]
Of course I mean re-simulate with different simulation parameters (Re-simulate setup). 

Typical cases: if a simulation seems to hang up, you cancel it and retry with a smaller StopTime. Or change the tolerance. Or change the solver, etc.

**Comment by adeas31** [comment]
> Of course I mean re-simulate with different simulation parameters (Re-simulate setup).
> 
> Typical cases: if a simulation seems to hang up, you cancel it and retry with a smaller StopTime. Or change the tolerance. Or change the solver, etc.

If you just want to change the simulation parameters then it is a lot easier. I will fix this asap.

Do you have a good MWE to test this?

**Comment by casella** [comment]
It turns out, the problem is not failed simulations, which is already possible to re-simulate. It is cancelled simulations, which is what we had in  #15166. Here are the MWEs.

If you open OMEdit, load this model and run it

```modelica
model Fails
  Real x;
equation
  der(x) = 1/(x - 0.5);
annotation(experiment(StopTime = 1));
end Fails;
```
it is possible to re-simulate it, e.g. with a different StopTime, without recompiling it.

Instead, if you load this model and run it, 
```modelica
model Hangs
  Real x(start = 1, fixed = true);
equation
  der(x) = noEvent(if x > 0 then -1 else 1);
  annotation(experiment(StopTime = 2));
end Hangs;
```
it hangs indefinitely. When you cancel the simulation, the Re-simulate buttons remain greyed out, so the only option to re-simulate it is to actually re-compile it. No big deal in this case, for a small MWE. Big deal if it take 2-3 minutes to re-compile, as in #15166.

**Comment by anotheruserofgithub** [comment]
> It turns out, the problem is not failed simulations, which is already possible to re-simulate. It is cancelled simulations

What about when it fails at initialization? You had an example model here: https://trac.openmodelica.org/OpenModelica/ticket/5472#comment:5. But I haven't tested yet.

Also I didn't check if the switch to the plotting perspective happens (or not) as expected, this seemed to be part of the discussion in https://trac.openmodelica.org/OpenModelica/ticket/5472, though it looks OK since the related ticket is closed:
- https://trac.openmodelica.org/OpenModelica/ticket/4935
- https://github.com/OpenModelica/OpenModelica/issues/4935

**Comment by casella** [comment]
Indeed, it is also not possible to re-simulate after initialization failure. Here's the MWE from the trac ticket:
```modelica
model TestInitialFailure
  Real x(start = 1);
equation
  der(x) = -x + 1;
initial equation
  x^2 + 1 = 0;
end TestInitialFailure;
```

**Comment by adeas31** [comment]
@casella I added the re-simulate button next to cancel compilation/simulation. Please try it out.

**Comment by casella** [comment]
> [@casella](https://github.com/casella) I added the re-simulate button next to cancel compilation/simulation. Please try it out.

Sorry, I don't get it. There is already a re-simulate button in the toolbar of the Plotting tab, which is greyed out

<img width="1616" height="1576" alt="Image" src="https://github.com/user-attachments/assets/8a41cd29-48f5-4c0a-b898-6455eb30bf86" />

Why should we add another button to perform the same action? We should still use the top toolbar button for consistency. 

Do I miss something?

**Comment by adeas31** [comment]
That re-simulate button is linked with Variable browser. It will re-simulate the selected item from the variable browser.

We could change that behavior. Instead of linking it to selected item in the variable browser we can link it to last simulated model. So whenever you click on re-simulate toolbar button it will re-simulate the last simulated model and if you want to re-simulate something from the variable browser you can right click on the item and choose re-simulate.

**Comment by casella** [comment]
> That re-simulate button is linked with Variable browser. It will re-simulate the selected item from the variable browser.

I see. I was not aware of that, I thought it was just the last simulated system 😅. That now explains why the button was greyed out: there were no results in the Variable browser to link to.

> We could change that behavior. Instead of linking it to selected item in the variable browser we can link it to last simulated model. So whenever you click on re-simulate toolbar button it will re-simulate the last simulated model and if you want to re-simulate something from the variable browser you can right click on the item and choose re-simulate.

Yes, this sounds a lot more intuitive and consistent with the overall GUI design. The Simulate and Simulation Setup refer to the last selected model, so it seems natural that Re-simulate applies to the last simulated (or attempted to simulate) models.

Good to have the option to re-simulate other models by right-click, that is also quite intuitive. 

BTW, could we also take the opportunity to implement #14111: when you open results, if the same directory also contains the simulation executable and _init.xml files, then it could could enable to re-simulate models that were compiled in a previous session, via right-click Re-simulate. Or even compiled models for which you just copied the simulation result directory into OMEdit's working directory, without even having access to the source code, as requested in #14111. @adeas31 what do you think?

**Comment by adeas31** [comment]
OK then I will update re-simulate toolbar buttons behavior.

For #14111, I will see what we can do. If we have the simulation executable and init.xml file then we can of course re-simulate.

**Comment by adeas31** [comment]
> OK then I will update re-simulate toolbar buttons behavior.

PR #15193 updated the re-simulate toolbar buttons behavior.
